File: 2605.从两个数字数组里生成最小数字/main.go
package leetcode

import "math"

func minNumber(nums1 []int, nums2 []int) int {
	min1 := math.Inf(1)
	min2 := math.Inf(1)
	minc := math.Inf(1)
	m := make(map[int]bool)

	for _, item := range nums1 {
		min1 = math.Min(min1, float64(item))

		m[item] = true
	}

	for _, item := range nums2 {
		min2 = math.Min(min2, float64(item))

		if _, ok := m[item]; ok {
			minc = math.Min(minc, float64(item))
		}
	}

	if minc < math.Inf(1) {
		return int(minc)
	}

	if min1 < min2 {
		return int(min1*10 + min2)
	} else {
		return int(min2*10 + min1)
	}
}

File: 2605.从两个数字数组里生成最小数字/main.ts
function minNumber(nums1: number[], nums2: number[]): number {
    let min1 = +Infinity;
    let min2 = +Infinity;
    const m = {};
    let minc = +Infinity;

    for(let i = 0; i < nums1.length;++i) {
        const item = nums1[i];
        min1 = Math.min(min1, item);

        m[item] = true;
    }
    for(let j = 0; j < nums2.length;++j) {
        const item = nums2[j];
        min2 = Math.min(min2, item);
        if(m[item]) minc = Math.min(minc, item);
    }

    if(minc < +Infinity) return minc;

    if(min1 < min2) return min1 * 10 + min2
    else return min2 * 10 + min1
};